test/forward_backward_compatibility/check_forward_backward_compatibility.py

```diff
@@ -139,6 +137,42 @@ def allow_listed(schema):
     ("dist_c10d", datetime.date(2099, 9, 17)),
 ]
 
+def has_valid_upgraders(schema, version_map):
+    # we want to parse through the map to find if
+    # the schema has valid upgraders. Since the
+    # version map has entry for each overload
+    # we need to do some ugly parsing.
+
+    # the name of the operator
+    schema_name = schema.name
+
+    # find if there are entries for this op
+    possible_overloads = []
+    possible_schemas = []
+    for key, upgrader_entries in version_map.items():
+        if key.split('.')[0] == schema_name:
+            possible_overloads.append(key)
+            possible_schemas.extend([entry.old_schema for entry in upgrader_entries])
+
+    # there is no entry for this schema
+    if len(possible_overloads) == 0:
+        return False
+
+    # let's make sure this existing schema is part of possible
+    # schemas
+    found = False
+    for old_schema in possible_schemas:
+        if parse_schema(old_schema) == schema:
+            found = True
+
+    if not found:
+        return False
+
+    current_version = torch._C._get_max_operator_version()
+    for overload in possible_overloads:
```

Comment:
Why do we need to check if if all possible_overloads is usable? Sounds like this isn't strictly necessary, is it because we can't tell which overload to match at this time?

Reply:
Yeah we can't tell which overload at this time, so we have to do it for all of them.